databricks/sdk/config.py

```diff
@@ -378,12 +381,22 @@ def _set_inner_config(self, keyword_args: Dict[str, any]):
             self.__setattr__(attr.name, keyword_args[attr.name])
 
     def _load_from_env(self):
+        if self.compute is None and self.cluster_id is None:
+            if (os.environ.get("DATABRICKS_COMPUTE") is not None
+                    and os.environ.get("DATABRICKS_CLUSTER_ID") is not None):
+                raise ValueError("Cannot specify both DATABRICKS_COMPUTE and "
+                                 "DATABRICKS_CLUSTER_ID in the environment.")
+
         found = False
         for attr in self.attributes():
             if not attr.env:
                 continue
             if attr.name in self._inner:
                 continue
+            if ((attr.name == "compute" and self.cluster_id is not None)
```

Comment:
I think this is going to be confusing as users may not be 100% familiar with the precedence between different options. Would it be better to require this to be unambiguous, i.e. either cluster_id or compute is set at exactly one place?